pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -87,6 +87,9 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		return err
 	}
 
+	// maintain the set's revision history limit
+	defer ssc.truncateHistory(set, pods, revisions, currentRevision, updateRevision)
```

Comment:
We are swallowing any errors returned from `truncateHistory` here. Not sure we can do much to handle the error, but I think it should at least be logged.

Reply:
Done in commit: https://github.com/kubernetes/kubernetes/pull/86097/commits/b0686ad740ef5e25705cac940a6e627ce81f26f9